python/cudf/bindings/copying.pyx

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from cudf.bindings.cudf_cpp cimport *
+from cudf.bindings.cudf_cpp import *
+from cudf.bindings.types cimport table as cudf_table
+from cudf.bindings.types import *
+from librmm_cffi import librmm as rmm
+from cudf.dataframe import columnops
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+pandas_version = tuple(map(int,pd.__version__.split('.', 2)[:2]))
+
+from librmm_cffi import librmm as rmm
+
+from libc.stdint cimport uintptr_t
+from libc.stdlib cimport calloc, malloc, free
+
+from libcpp.map cimport map as cmap
+from libcpp.string  cimport string as cstring
+
+
+cdef cudf_table* table_from_cols(cols):
+    col_count=len(cols)
+    cdef gdf_column **c_cols = <gdf_column**>malloc(sizeof(gdf_column*)*col_count)
```

Comment:
It looks like this is never freed. You could use a `std::vector` via Cython to make this cleaner: https://cython.readthedocs.io/en/latest/src/userguide/wrapping_CPlusPlus.html#standard-library

Reply:
cudf::table has the reference of c_cols,
I've changed to return gdf_column** instead and free it at same time to free table.